packet/cloud.go

```diff
@@ -176,47 +176,41 @@ func (c *cloud) HasClusterID() bool {
 func startNodesWatcher(informer informers.SharedInformerFactory, handlers []nodeReconciler, stop <-chan struct{}) error {
 	klog.V(5).Info("called startNodesWatcher")
 	if len(handlers) == 0 {
-		klog.V(5).Info("no service handlers to process")
+		klog.V(5).Info("no node handlers to process")
 		return nil
 	}
-	nodes := informer.Core().V1().Nodes()
-	nodesLister := nodes.Lister()
-
-	// next make sure existing nodes have it set
-	nodeList, err := nodesLister.List(labels.Everything())
-	if err != nil {
-		return fmt.Errorf("failed to list nodes: %v", err)
-	}
-	klog.V(2).Infof("existing nodes: %v", nodeList)
-	for _, h := range handlers {
-		if err := h(nodeList, false); err != nil {
-			klog.Errorf("failed to update and sync nodes for handler: %v", err)
-		}
-	}
 
 	klog.V(5).Info("startNodesWatcher(): creating nodesInformer")
-	nodesInformer := nodes.Informer()
+	nodesInformer := informer.Core().V1().Nodes().Informer()
 	klog.V(5).Info("startNodesWatcher(): adding event handlers")
 	nodesInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			n := obj.(*v1.Node)
 			for _, h := range handlers {
-				if err := h([]*v1.Node{n}, false); err != nil {
+				if err := h([]*v1.Node{n}, ModeAdd); err != nil {
 					klog.Errorf("failed to update and sync node for add %s for handler: %v", n.Name, err)
 				}
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			n := obj.(*v1.Node)
 			for _, h := range handlers {
-				if err := h([]*v1.Node{n}, false); err != nil {
+				if err := h([]*v1.Node{n}, ModeRemove); err != nil {
 					klog.Errorf("failed to update and sync node for remove %s for handler: %v", n.Name, err)
 				}
 			}
 		},
 	})
+
 	klog.V(5).Info("startNodesWatcher(): nodesInformer.Run()")
 	go nodesInformer.Run(stop)
+	syncFuncs := []cache.InformerSynced{
```

Comment:
I had to read up on this to figure out what was going on here.
A comment could help future reviews.

https://engineering.bitnami.com/articles/a-deep-dive-into-kubernetes-controllers.html

Reply:
Definitely. I will open a new PR with better documentation.

FWIW, _every time_ I go near this, I need to read up again. 😂

Reply:
See #97 @displague , just for you. :-)